dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/utils/ProgramBuilder.java

```diff
@@ -346,4 +351,14 @@ public void addSwwPairThreads(int threadId0, int threadId1) {
             thread0.getSyncSet().add(thread1);
         }
     }
+
+    // ----------------------------------------------------------------------------------------------------------------
+    // OpenCL
+    public void setReg2LocMap(Register reg, MemoryObject loc) {
+        reg2LocMap.put(reg, loc);
+    }
+
+    public MemoryObject getLocFromReg(Register reg) {
+        return reg2LocMap.get(reg);
+    }
```

Comment:
Where is this being used? If it is used for some initialization of the form `r0:&a` can we somehow create some local events at the beginning of the corresponding thread with `r0 <- &a`. I think this is what we are doing with threads having parameters of the form `P0(atomic_t *x, atomic_t *y)` ... they are transformed to `x <- &x` where the left hand side is a register.